src/main/primitives/Colour.java

```diff
@@ -41,21 +44,49 @@ public Colour add(Colour colour) {
 	 * @return The new colour which is the result of the scale.
 	 */
 	public Colour scale(double factor) {
-		// TODO: implement
-		return null;
+		return new Colour(this.rgb.scale(factor));
+	}
+
+	/**
+	 * Get the integer value of red of this colour 
+	 * @return The integer value (0-255) of red
+	 */
+	public int red() {
+		return getInt(rgb.x);
+	}
+
+	/**
+	 * Get the integer value of green of this colour 
+	 * @return The integer value (0-255) of green
+	 */
+	public int green() {
+		return getInt(rgb.y);
 	}
 
 	/**
-	 * Returns the RGB value representing the color. (Bits 24-31 are alpha, 16-23 are red, 8-15 are green, 0-7 are
-	 * blue).
+	 * Get the integer value of blue of this colour 
+	 * @return The integer value (0-255) of blue
+	 */
+	public int blue() {
+		return getInt(rgb.z);
+	}
+
+	/**
+	 * Returns the RGB value representing the color. (Bits 24-31 are blank since alpha is unused, 16-23 are red, 8-15 are green, 0-7 are
+	 * blue). The imageType will be TYPE_INT_RGB
 	 *
 	 * @return the RGB value of the color.
 	 */
 	public int rgb() {
-		// BufferedImage, as used by ImageWriter, needs only the RGB value (int) to write a pixel. So no point exposing
-		// the actual java colour. That is just implementation detail (if we want to use it... (bitwise operations are
-		// fun)).
-		// TODO: implement
-		return 0;
+		// TODO: Check if this messes up all of our images
+		return (red() << 16) ^ (green() << 8) ^ blue();
+	}
+
+	private int getInt(double value) {
+		if (value < 0)
+			return 0;
+		if (value > 255)
+			return 255;
+		return (int) Math.round(value);
```

Comment:
Good spot! I would have just used `(int) value`, forgetting to round to closest XD